profiling/src/exporter/mod.rs

```diff
@@ -130,30 +145,55 @@ impl ProfileExporter {
     ) -> anyhow::Result<Request> {
         let mut form = multipart::Form::default();
 
-        form.add_text("version", "3");
-        form.add_text("start", start.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string());
-        form.add_text("end", end.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string());
-        form.add_text("family", self.family.as_ref());
-
-        for tags in self.tags.as_ref().iter().chain(additional_tags.iter()) {
-            for tag in tags.iter() {
-                form.add_text("tags[]", tag.to_string());
-            }
+        // combine tags and additional_tags
+        let mut tags_profiler = String::new();
+        let other_tags = additional_tags.into_iter();
+        for tag in self.tags.iter().chain(other_tags).flatten() {
+            tags_profiler.push_str(tag.as_ref());
+            tags_profiler.push(',');
         }
+        tags_profiler.pop(); // clean up the trailing comma
+
+        let attachments: Vec<String> = files.iter().map(|file| file.name.to_owned()).collect();
+
+        let event = json!({
+            "attachments": attachments,
+            "tags_profiler": tags_profiler,
+            "start": start.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string(),
+            "end": end.format("%Y-%m-%dT%H:%M:%S%.9fZ").to_string(),
+            "family": self.family.as_ref(),
+            "version": "4",
+        })
+        .to_string();
+
+        form.add_reader_file_with_mime(
+            // intake will look for a field of this name
```

Comment:
Should this be file, not field?

```suggestion
            // intake will look for a file of this name
```

Reply:
Interestingly, it wasn't a mistake. Intake does not look for `filename=event.json`, it looks for `name=event`.

Reply:
Ahhh. I guess `// intake looks for name=event for this file` or something similar would be clearer? Or even `// Intake does not look for filename=event.json, it looks for name=event`? 😄